writable.go

```diff
@@ -18,6 +18,13 @@ func BytesWritable(b []byte) []byte {
 	return b[4:]
 }
 
+func PutBytesWritable(raw []byte) []byte {
```

Comment:
so in the case of `encoding/binary`'s `Put*` methods, you pass in the byte slice but it already has to be of a size large enough to hold the value being placed in: https://github.com/kitchen/gokitchentest/blob/master/putuint32_test.go but those are also all fixed length representations, so you can know them in advance. However, I don't know if that is the case with the other Writables, or if that's even an expected property of them. (I also don't understand exactly why `encoding/binary` blows up if you give it a byte slice that isn't already the correct size, even if it has been allocated to be large enough.)